service/frontend/configs/quotas.go

```diff
@@ -48,53 +48,79 @@ var (
-		"RecordActivityTaskHeartbeatById":  2,
-		"RespondActivityTaskCanceled":      2,
-		"RespondActivityTaskCanceledById":  2,
-		"RespondActivityTaskFailed":        2,
-		"RespondActivityTaskFailedById":    2,
-		"RespondActivityTaskCompleted":     2,
-		"RespondActivityTaskCompletedById": 2,
-		"RespondWorkflowTaskCompleted":     2,
-		"RespondWorkflowTaskFailed":        2,
-		"RespondQueryTaskCompleted":        2,
-
-		// priority 3
-		"ResetWorkflowExecution":             3,
-		"DescribeWorkflowExecution":          3,
-		"QueryWorkflow":                      3,
-		"PollWorkflowTaskQueue":              3,
-		"PollActivityTaskQueue":              3,
-		"PollWorkflowExecutionUpdate":        3,
-		"GetWorkflowExecutionHistoryReverse": 3,
-		"GetWorkerBuildIdCompatibility":      3,
-		"GetWorkerTaskReachability":          3,
-		"DeleteWorkflowExecution":            3,
-
-		// priority 4
-		"ResetStickyTaskQueue":    4,
-		"DescribeTaskQueue":       4,
-		"ListTaskQueuePartitions": 4,
+		"CreateSchedule":                   1,
+		"StartBatchOperation":              1,
+
+		// P2: Change State APIs
+		"RequestCancelWorkflowExecution": 2,
+		"TerminateWorkflowExecution":     2,
+		"ResetWorkflowExecution":         2,
+		"DeleteWorkflowExecution":        2,
+		"GetWorkflowExecutionHistory":    2, // relatively high priority because it is required for replay
+		"UpdateSchedule":                 2,
+		"PatchSchedule":                  2,
+		"DeleteSchedule":                 2,
+		"StopBatchOperation":             2,
+
+		// P3: Status Querying APIs
+		"DescribeWorkflowExecution":     3,
+		"DescribeTaskQueue":             3,
+		"GetWorkerBuildIdCompatibility": 3,
+		"GetWorkerTaskReachability":     3,
+		"ListTaskQueuePartitions":       3,
+		"QueryWorkflow":                 3,
+		"DescribeSchedule":              3,
+		"ListScheduleMatchingTimes":     3,
+		"ListSchedules":                 3,
+		"DescribeBatchOperation":        3,
+		"ListBatchOperations":           3,
+
+		// P4: Progress APIs
+		"RecordActivityTaskHeartbeat":      4,
+		"RecordActivityTaskHeartbeatById":  4,
+		"RespondActivityTaskCanceled":      4,
+		"RespondActivityTaskCanceledById":  4,
+		"RespondActivityTaskFailed":        4,
+		"RespondActivityTaskFailedById":    4,
+		"RespondActivityTaskCompleted":     4,
+		"RespondActivityTaskCompletedById": 4,
+		"RespondWorkflowTaskCompleted":     4,
+		"RespondWorkflowTaskFailed":        4,
+		"RespondQueryTaskCompleted":        4,
+		"RespondNexusTaskCompleted":        4,
+		"RespondNexusTaskFailed":           4,
+
+		// P5: Poll APIs and other low priority APIs
+		"PollWorkflowTaskQueue":              5,
+		"PollActivityTaskQueue":              5,
+		"PollWorkflowExecutionUpdate":        5,
+		"PollNexusTaskQueue":                 5,
+		"ResetStickyTaskQueue":               5,
+		"GetWorkflowExecutionHistoryReverse": 5,
 	}
 
-	ExecutionAPIPrioritiesOrdered = []int{0, 1, 2, 3, 4}
+	ExecutionAPIPrioritiesOrdered = []int{0, 1, 2, 3, 4, 5}
```

Comment:
If all our priorities are ordered why do we need this?

Reply:
Good question. Probably just need MaxPriority value. But will leave it as is in this PR.